src/communication/xcSession.ts

```diff
@@ -167,12 +171,14 @@ export class DefaultSession implements Session {
         }, this);
         clearInterval(this.heartbeatTimer);
         this.dispose();
+        this.closedByUser = true;
         this.webSocket.close();
     };
 
 }
 
-export const SessionFactory = function (serverUrl: string, configuration: ApiConfiguration, sessionData: string): Session {
+export const SessionFactory = (serverUrl: string, configuration: ApiConfiguration, sessionData: string): Session => {
+    const WebSocket: any = javascriptHelper().WebSocket;
```

Comment:
do you really need a constant? 
does `webSocket = new javascriptHelper().WebSocket(serverUrl);` work?
BTW, do you really need a class here? the helper could give you an instance, like:
`webSocket = javascriptHelper().createWebSocket(serverUrl);`